core/core-impl/src/main/java/org/openbaton/nfvo/core/api/NetworkServiceRecordManagement.java

```diff
@@ -478,17 +473,27 @@ public void addVNFCInstance(String id, String idVnf, VNFComponent component, Str
       throw new UnauthorizedUserException(
           "VNFR not under the project chosen, are you trying to hack us? Just kidding, it's a bug :)");
     }
-    VirtualDeploymentUnit virtualDeploymentUnit =
-        virtualNetworkFunctionRecord.getVdu().iterator().next();
-    if (virtualDeploymentUnit == null) {
-      throw new NotFoundException("No VirtualDeploymentUnit found");
+
+    List<VirtualDeploymentUnit> vdusFound = new LinkedList<>();
+    VirtualDeploymentUnit virtualDeploymentUnit = null;
+
+    for (VirtualDeploymentUnit vdu : virtualNetworkFunctionRecord.getVdu()) {
+      if (vdu.getProjectId() != null && vdu.getProjectId().equals(projectId)) vdusFound.add(vdu);
     }
 
-    if (virtualDeploymentUnit.getScale_in_out()
-        == virtualDeploymentUnit.getVnfc_instance().size()) {
-      throw new WrongStatusException(
-          "The VirtualDeploymentUnit chosen has reached the maximum number of VNFCInstance");
+    if (vdusFound.size() == 0) throw new NotFoundException("No VirtualDeploymentUnit found");
+
+    for (VirtualDeploymentUnit vdu : vdusFound) {
+      if (vdu.getScale_in_out() == vdu.getVnfc_instance().size()) continue;
```

Comment:
What happens if vdu.getVnfc_instance().size() is less than vdu.getScale_in_out()? Should also be scalable in this case, shouldn't it?